tests/mtouch/ToolTasksBinPathTest.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Diagnostics;
+using NUnit.Framework;
+using Xamarin.Tests;
+
+namespace AppleSdkSettings {
+
+	public static class XcodeVersion {
+		public static int Major = Version.Parse (Configuration.XcodeVersion).Major;
+	}
+}
+
+namespace Xamarin.MacDev.Tasks {
+
+	public abstract class MetalTaskBase {
+
+		protected abstract string OperatingSystem {
+			get;
+		}
+
+		protected abstract string DevicePlatformBinDir {
+			get;
+		}
+
+		protected string SdkDevPath {
+			get { return ""; }
+		}
+	}
+
+	public abstract class MetalLibTaskBase {
+
+		protected abstract string DevicePlatformBinDir {
+			get;
+		}
+
+		protected string SdkDevPath {
+			get { return ""; }
+		}
+	}
+}
+
+namespace Xamarin.Mac.Tasks {
+
+	public class MetalPoker : Metal {
+
+		public new string DevicePlatformBinDir {
+			get { return base.DevicePlatformBinDir; }
+		}
+	}
+
+	public class MetalLibPoker : MetalLib {
+
+		public new string DevicePlatformBinDir {
+			get { return base.DevicePlatformBinDir; }
+		}
+	}
+
+	[TestFixture]
+	public class ToolTasksBinPathTest {
+
+		[SetUp]
+		public void Setup ()
+		{
+			Environment.SetEnvironmentVariable ("DEVELOPER_DIR", Configuration.xcode_root);
```

Comment:
that change the execution environment, i.e. it could affect other (existing or future) tests randomly (since the execution order is not guaranteed) and that will be hard to diagnose

To avoid this you can move this into a 
```
var dd = getenv ("DEVELOPER_DIR");
try {
   setenv ("DEVELOPER_DIR", Configuration.xcode_root);
   xcrun (...);
}
finally {
   setenv ("DEVELOPER_DIR", dd);
}
```

Reply:
Or you could set the environment variable in the `ProcessStartInfo` instead, then you wouldn't need any cleanup.